components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowConsumer.java

```diff
@@ -183,6 +186,7 @@ private void sendResponse(HttpServerExchange httpExchange, Exchange camelExchang
 
         if (body == null) {
             log.trace("No payload to send as reply for exchange: {}", camelExchange);
+            this.changeStatusCodeToNoContent(camelExchange, httpExchange);
```

Comment:
Do you think as well, if we handle the status code from here instead: https://github.com/apache/camel/blob/master/components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHttpBinding.java#L314? From the top of my head, if you handle it from there (which is the main responsibility of that method), you won't need any more modifications elsewhere apart from the tests. Also, I'd recommend to use the enum `StatusCode.NO_CONTENT` instead of the raw int

Reply:
After going through the debugger, wasn't sure of the best place to implement it. There were quite a few classes. Was hoping for that feedback. I'll try moving the code for this feature and test it.

Reply:
Sorry here: https://github.com/apache/camel/blob/master/components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHttpBinding.java#L325

Reply:
Looks like moving it to `DefaultUndertowHttpBinding` is working! Thanks

Reply:
You welcome! Also I advice to do the same in your other PR :)